### PR TITLE
Slabs and stairs

### PR DESCRIPTION
**Pull request by namebee**
This is a draft, Do not merge yet. A few things are broken.

### CODE REVIEW

**Review by dzwdz — APPROVED** [review]

File: src/main/java/io/github/hydos/proton/base/block/ProtonSlabBlock.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    public final Block parent;
+
+    public ProtonSlabBlock(Block parent) {
+        super(VariantHandler.realStateCopy(parent));
+
+        this.parent = parent;
+        ProtonRegisterUtil.block(new Identifier(Registry.BLOCK.getId(parent).toString() + "_slab"), this);
+        ProtonRegisterUtil.item(new Identifier(Registry.BLOCK.getId(parent).toString() + "_slab"), new BlockItem(this, new Item.Settings().group(ItemGroup.BUILDING_BLOCKS)));
+
+        // TODO: uhh
```
Comment (hYdos) [inline src/main/java/io/github/hydos/proton/base/block/ProtonSlabBlock.java]: finish the TODO :tinypotato:

File: src/main/java/io/github/hydos/proton/base/block/ProtonStairsBlock.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    public final Block parent;
+
+    public ProtonStairsBlock(Block parent) {
+        super(parent.getDefaultState(), VariantHandler.realStateCopy(parent));
+
+        this.parent = parent;
+        ProtonRegisterUtil.block(new Identifier(Registry.BLOCK.getId(parent).toString() + "_stairs"), this);
+        ProtonRegisterUtil.item(new Identifier(Registry.BLOCK.getId(parent).toString() + "_stairs"), new BlockItem(this, new Item.Settings().group(ItemGroup.BUILDING_BLOCKS)));
+
+        // TODO: uhh
```
Comment (hYdos) [inline src/main/java/io/github/hydos/proton/base/block/ProtonStairsBlock.java]: finish thank

File: src/main/java/io/github/hydos/proton/base/block/ProtonWallBlock.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    public ProtonWallBlock(Block parent) {
+        super(VariantHandler.realStateCopy(parent));
+
+        this.parent = parent;
+        ProtonRegisterUtil.block(new Identifier(Registry.BLOCK.getId(parent).toString() + "_wall"), this);
+        ProtonRegisterUtil.item(new Identifier(Registry.BLOCK.getId(parent).toString() + "_wall"), new BlockItem(this, new Item.Settings().group(ItemGroup.BUILDING_BLOCKS)));
+
+        // TODO: uhh
+        //RenderLayerHandler.setInherited(this, parent.getBlock());
```
Comment (hYdos) [inline src/main/java/io/github/hydos/proton/base/block/ProtonWallBlock.java]: you know what im gonna say at this point

**Review by hYdos — CHANGES_REQUESTED** [review]

File: src/main/resources/assets/proton/blockstates/thatch_stairs.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    "facing=east,half=top,shape=inner_right":  { "model": "proton:block/thatch_stairs_inner", "x": 180, "y": 90, "uvlock": true },
+    "facing=west,half=top,shape=inner_right":  { "model": "proton:block/thatch_stairs_inner", "x": 180, "y": 270, "uvlock": true },
+    "facing=south,half=top,shape=inner_right": { "model": "proton:block/thatch_stairs_inner", "x": 180, "y": 180, "uvlock": true },
+    "facing=north,half=top,shape=inner_right": { "model": "proton:block/thatch_stairs_inner", "x": 180, "uvlock": true },
+    "facing=east,half=top,shape=inner_left":  { "model": "proton:block/thatch_stairs_inner", "x": 180, "uvlock": true },
+    "facing=west,half=top,shape=inner_left":  { "model": "proton:block/thatch_stairs_inner", "x": 180, "y": 180, "uvlock": true },
+    "facing=south,half=top,shape=inner_left": { "model": "proton:block/thatch_stairs_inner", "x": 180, "y": 90, "uvlock": true },
+    "facing=north,half=top,shape=inner_left": { "model": "proton:block/thatch_stairs_inner", "x": 180, "y": 270, "uvlock": true }
+  }
+}
```
Comment (YTG1234) [inline src/main/resources/assets/proton/blockstates/thatch_stairs.json:44]: I don't know much about blockstate variants and such, I'll have to trust you that you did it right

**Review by YTG1234 — APPROVED** [review]
Nice PR

**Review by hYdos — APPROVED** [review]

**Pull request merged**